injector/container_failure.go

```diff
@@ -35,18 +35,18 @@ func NewContainerFailureInjector(spec v1beta1.ContainerFailureSpec, config Conta
 		config.ProcessManager = process.NewManager(config.DryRun)
 	}
 
-	return containerFailureInjector{
+	return &containerFailureInjector{
```

Comment:
On all injectors:
- added the method UpdateConfig to be able to reset it when the container changes PID and needs to reset the managers
- transformed the injector struct on all methods to a pointer to be able to keep the conf updated when resetting it